jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/UpdateChecker.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.plugins.common;
+
+import com.google.cloud.tools.jib.filesystem.XdgDirectories;
+import com.google.cloud.tools.jib.json.JsonTemplate;
+import com.google.cloud.tools.jib.json.JsonTemplateMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+/** Checks if Jib is up-to-date. */
+public class UpdateChecker {
+
+  /** JSON template for the configuration file used to enable/disable update checks. */
+  @VisibleForTesting
+  static class ConfigJsonTemplate implements JsonTemplate {
+    private boolean disableUpdateCheck;
+
+    @VisibleForTesting
+    void setDisableUpdateCheck(boolean disableUpdateCheck) {
+      this.disableUpdateCheck = disableUpdateCheck;
+    }
+  }
+
+  /**
+   * Begins checking for an update in a separate thread.
+   *
+   * @param skip if {@code true}, the update check itself will be skipped
+   * @param version the current version of Jib being used
+   * @param versionUrl the location to check for the latest version
+   * @param executorService the {@link ExecutorService}
+   * @return a new {@link UpdateChecker}
+   */
+  public static UpdateChecker checkForUpdate(
+      boolean skip, String version, String versionUrl, ExecutorService executorService) {
```

Comment:
It feels weird to put "skip" in here?

Could the maven/gradle calls to checkForUpdate be implemented in a way like

```
Updater.for(ProjectProperties).checkForUpdates()
```
where "for" decides whether to skip or not? Or something of that sort.